Signal/src/ViewControllers/SendExternalFileViewController.m

```diff
@@ -0,0 +1,146 @@
+//
+//  Copyright (c) 2017 Open Whisper Systems. All rights reserved.
+//
+
+#import "SendExternalFileViewController.h"
+#import "Environment.h"
+#import "Signal-Swift.h"
+#import "ThreadUtil.h"
+#import "UIColor+OWS.h"
+#import "UIFont+OWS.h"
+#import <SignalServiceKit/OWSMessageSender.h>
+#import <SignalServiceKit/TSThread.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface SendExternalFileViewController () <SelectThreadViewControllerDelegate>
+
+@property (nonatomic, readonly) OWSMessageSender *messageSender;
+
+@end
+
+#pragma mark -
+
+@implementation SendExternalFileViewController
+
+- (instancetype)init
+{
+    if (self = [super init]) {
+        self.delegate = self;
+    }
+    return self;
+}
+
+- (void)loadView
+{
+    [super loadView];
+
+    _messageSender = [Environment getCurrent].messageSender;
+
+    self.title = NSLocalizedString(@"SEND_EXTERNAL_FILE_VIEW_TITLE", @"Title for the 'send external file' view.");
+}
+
+#pragma mark - SelectThreadViewControllerDelegate
+
+- (void)threadWasSelected:(TSThread *)thread
+{
+    OWSAssert(self.attachment);
+    OWSAssert(thread);
+
+    [ThreadUtil sendMessageWithAttachment:self.attachment inThread:thread messageSender:self.messageSender];
+
+    [Environment messageThreadId:thread.uniqueId];
+}
+
+- (BOOL)canSelectBlockedContact
+{
+    return NO;
+}
+
+- (nullable UIView *)createHeader:(UIView *)superview
+{
+    OWSAssert(superview)
+
+        const CGFloat imageSize
+        = ScaleFromIPhone5To7Plus(40, 40);
+    const CGFloat imageLabelSpacing = ScaleFromIPhone5To7Plus(5, 5);
```

Comment:
If I had to guess, I'd say this is a convention to communicate that you've considered iPhone5->7+ in your layout without actually changing any behavior?